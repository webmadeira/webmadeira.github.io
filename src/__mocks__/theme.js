export default {
  color() { return '#000' },
  contrast() { return 'rgb(255, 255, 255)' },
  textSetting() {
    return {
      fontSize: '12px',
      lineHeight: '1.2rem',
    }
  },
  weight() { return 100 },
  fontFamily() { return 'Roboto Mono' },
}
