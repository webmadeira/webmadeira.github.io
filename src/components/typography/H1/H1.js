import styled from 'styled-components'

const H1 = styled.h1`
  font-family: ${props => props.theme.fontFamily('secondary')};
  font-size: 4.1rem;
  line-height: 5rem;
`

export default H1
