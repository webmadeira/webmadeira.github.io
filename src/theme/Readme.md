# Theme

In this section you can find all the resources related to webmadeira's theme provider.  
All this resources are available across all the `src` code through the property `theme`.  
Read more about [Theming with `styled-components`](https://www.styled-components.com/docs/advanced).

## Color palettes

Color palettes provide you all the colors you need to build your UIs.

At `theme/color-palettes` you can find all the existing palettes and variations and you can preview them at "Theme/Color palettes" section at storybook.

### Usage

These palettes are available on your stylings as follows:

~~~js
import styled from 'styled-components'

const StyledComponent = styled.div`
  background-color: ${props => props.theme.color('[palette]', '[variant]')}
`

const MyComponent = () => (
  <StyledComponent />
)

export default MyComponent
~~~

Where **[palette]** and **[variant]** are then name of the palette and which variant within the palette you want to use.

**Note**: Theme **[palette]** and **[variant]** parameters defaults to `'primary'` and `'base'`.

## Typography

For typography we have `textSetting`, `weight` and `fontFamily` resources.

### Text setting

**`textSetting`** is a function that can be used as `theme.textSetting([size])`, where `[size]` can assume the following values:

* `'xs'`
* `'sm'`
* `'nr'`
* `'md'`
* `'lg'`
* `'xl'`

And as result you get an object with the following shape:  
~~~js
{
  fontSize: '[X]rem', // 18px
  lineHeight: '[X]rem',
}
~~~

### Font weight

**`weight`** is a function that can be used as `theme.weight([weight])`, where `[weight]` can assume the following values:

* `'light'`
* `'regular'`
* `'medium'`
* `'bold'`

And as result you the get the amount of **weight** to apply to your font (e.g. `300` or `400`).

### Font family

`fontFamily` is a function that can be used as `theme.fontFamily([type])`, where `[type]` can assume the following values:

* `'primary'`
* `'secondary'`

And as result you the fontface to apply on your *font-family* CSS rule.

## Mixins

Mixins in this context are JS functions that return CSS that might be used along with other styles. Check `theme/mixins` to see all the existing mixins.

### Usage

~~~js
import styled from 'styled-components'

const StyledComponent = styled.div`
  padding: 10px;
  font-size: ${props => props.theme.textSetting('xs').fontSize};
  ${props => props.theme.mixins.[MY_MIXIN]()}
`

const MyComponent = () => (
  <StyledComponent />
)

export default MyComponent
~~~