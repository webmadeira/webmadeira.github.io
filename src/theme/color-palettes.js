const palettes = {
  brand: {
    base: '#ed4042',
  },
  primary: {
    base: '#14455a',
  },
  secondary: {
    dark: '#65abb7',
    base: '#c1e9f0',
    light: '#d9e2e3',
    bright: '#45e4ff',
  },
  neutral: {
    dark: '#000',
    base: '#535353',
    bright: '#fff',
  },
}

const transparencies = {
  dark: {
    primary: 'rgb(255, 255, 255)',
    secondary: 'rgba(255, 255, 255, .70)',
    disabled: 'rgba(255, 255, 255, .5)',
    hint: 'rgba(255, 255, 255, .5)',
  },
  light: {
    primary: 'rgba(0, 0, 0, .87)',
    secondary: 'rgba(0, 0, 0, .54)',
    disabled: 'rgba(0, 0, 0, .38)',
    hint: 'rgba(0, 0, 0, .38)',
  },
}

export {
  palettes,
  transparencies,
}
