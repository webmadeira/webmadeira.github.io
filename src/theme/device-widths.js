export default {
  giant: 1170,
  desktop: 992,
  tablet: 768,
  phone: 480,
  iphone: 376,
}
