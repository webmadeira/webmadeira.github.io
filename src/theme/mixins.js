// these sizes are arbitrary and you can set them to whatever you wish
import { css } from 'styled-components'

const sizes = {
  giant: 1170,
  desktop: 992,
  tablet: 768,
  phone: 480,
  iphone: 376,
}

// iterate through the sizes and create a media template
const mediaQuery = Object.keys(sizes).reduce((acc, label) => {
  const emSize = sizes[label] / 16
  const accumulator = Object.assign(
    {},
    acc,
    {
      [label]: (...args) => css`
        @media (max-width: ${emSize}em) {
          ${css(...args)}
        }
      `,
    },
  )

  return accumulator
}, {})


const center = () => (`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`)

export {
  center,
  mediaQuery,
}
