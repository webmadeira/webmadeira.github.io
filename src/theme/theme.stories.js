import React from 'react'
import { storiesOf } from '@storybook/react'
import withReadme from 'storybook-readme/with-readme'
import styled from 'styled-components'
import Readme from './Readme.md'
import { H3, Body2, Caption } from '../components/typography'

const sharedStyle = theme => `
  color: ${theme.contrast('dark', 'secondary')};
  padding: 60px;
  text-align: center;
  flex-grow: 1;
  flex-basis: 0;
  border: 1px solid ${theme.color('neutral', 'bright')};
`

const BrandBasePalette = styled.div`
  background-color: ${props => props.theme.color('brand')};
  ${props => sharedStyle(props.theme)}
`

const PrimaryBasePalette = styled.div`
  background-color: ${props => props.theme.color()};
  ${props => sharedStyle(props.theme)}
`

const SecondaryDarkPalette = styled.div`
  background-color: ${props => props.theme.color('secondary', 'dark')};
  ${props => sharedStyle(props.theme)}
`

const SecondaryBasePalette = styled.div`
  background-color: ${props => props.theme.color('secondary', 'base')};
  ${props => sharedStyle(props.theme)}
  color: ${props => props.theme.contrast('light', 'secondary')};
`

const SecondaryLightPalette = styled.div`
  background-color: ${props => props.theme.color('secondary', 'light')};
  ${props => sharedStyle(props.theme)}
  color: ${props => props.theme.contrast('light', 'secondary')};
`

const SecondaryBrightPalette = styled.div`
  background-color: ${props => props.theme.color('secondary', 'bright')};
  ${props => sharedStyle(props.theme)}
  color: ${props => props.theme.contrast('light', 'secondary')};
`

const NeutralDarkPalette = styled.div`
  background-color: ${props => props.theme.color('neutral', 'dark')};
  ${props => sharedStyle(props.theme)}
`

const NeutralBasePalette = styled.div`
  background-color: ${props => props.theme.color('neutral')};
  ${props => sharedStyle(props.theme)}
`

const NeutralBrightPalette = styled.div`
  background-color: ${props => props.theme.color('neutral', 'bright')};
  ${props => sharedStyle(props.theme)}
  color: ${props => props.theme.contrast('light', 'secondary')};
`

const Palettes = () => (
  <div style={{ display: 'flex', flexWrap: 'wrap' }}>
    <BrandBasePalette>
      <H3>Brand palette</H3>
      <Caption>base</Caption>
    </BrandBasePalette>
    <PrimaryBasePalette>
      <H3>Primary palette</H3>
      <Caption>base</Caption>
    </PrimaryBasePalette>
    <SecondaryDarkPalette>
      <H3>Secondary palette</H3>
      <Caption>dark</Caption>
    </SecondaryDarkPalette>
    <SecondaryBasePalette>
      <H3>Secondary palette</H3>
      <Caption>base</Caption>
    </SecondaryBasePalette>
    <SecondaryLightPalette>
      <H3>Secondary palette</H3>
      <Caption>light</Caption>
    </SecondaryLightPalette>
    <SecondaryBrightPalette>
      <H3>Secondary palette</H3>
      <Caption>bright</Caption>
    </SecondaryBrightPalette>
    <NeutralDarkPalette>
      <H3>Neutral palette</H3>
      <Caption>dark</Caption>
    </NeutralDarkPalette>
    <NeutralBasePalette>
      <H3>Neutral palette</H3>
      <Caption>base</Caption>
    </NeutralBasePalette>
    <NeutralBrightPalette>
      <H3>Neutral palette</H3>
      <Caption>bright</Caption>
    </NeutralBrightPalette>
  </div>
)

const CheckReadme = () => (
  <Body2>Check the Readme.md for more details</Body2>
)

storiesOf('Theme', module)
  .addDecorator(withReadme(Readme))
  .add('Color palettes', () => (<Palettes />))
  .add('Typography', CheckReadme)
