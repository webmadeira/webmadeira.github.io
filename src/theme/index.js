import { palettes, transparencies } from './color-palettes'
import { weights, textSettings, fontFamilies } from './typography'
import * as mixins from './mixins'

function color(level = 'primary', variant = 'base') {
  return palettes[level][variant]
}

function contrast(background = 'light', type = 'primary') {
  return transparencies[background][type]
}

function textSetting(level) {
  return textSettings[level]
}

function weight(level) {
  return weights[level]
}

function fontFamily(key) {
  return fontFamilies[key]
}

export {
  color,
  contrast,
  textSetting,
  weight,
  fontFamily,
  mixins,
}
