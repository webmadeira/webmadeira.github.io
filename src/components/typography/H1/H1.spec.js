import React from 'react'
import { shallow } from 'enzyme'
import H1 from './H1'
import theme from '../../../__mocks__/theme'

describe('Typography - H1 component', () => {
  it('must render properly', () => {
    const component = shallow(<H1 theme={theme}>H1</H1>)

    expect(component).toMatchSnapshot()
  })
})
