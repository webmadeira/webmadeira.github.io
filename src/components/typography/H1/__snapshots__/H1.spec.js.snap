// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography - H1 component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.h1
    theme={
      Object {
        "color": [Function],
        "contrast": [Function],
        "fontFamily": [Function],
        "textSetting": [Function],
        "weight": [Function],
      }
    }
  >
    H1
  </styled.h1>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "H1",
      "className": "sc-bdVaJa puwh",
    },
    "ref": null,
    "rendered": "H1",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "H1",
        "className": "sc-bdVaJa puwh",
      },
      "ref": null,
      "rendered": "H1",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
